internal/command/validate.go

```diff
@@ -64,30 +66,73 @@ func (c *ValidateCommand) Run(rawArgs []string) int {
 	return view.Results(diags)
 }
 
-func (c *ValidateCommand) validate(dir string) tfdiags.Diagnostics {
+func (c *ValidateCommand) validate(dir, testDir string, noTests bool) tfdiags.Diagnostics {
 	var diags tfdiags.Diagnostics
 
-	cfg, cfgDiags := c.loadConfig(dir)
+	cfg, cfgDiags := c.loadConfigWithTests(dir, testDir)
```

Comment:
Is it possible that this acts differently than `loadConfig` when the user wants `-no-tests`?

Reply:
Yep, good shout. I've made it use `loadConfig` when `-no-tests` is set.